homeassistant/components/melnor/sensor.py

```diff
@@ -0,0 +1,133 @@
+
+from .const import DOMAIN
+from .models import MelnorBluetoothBaseEntity, MelnorDataUpdateCoordinator
+
+
+@dataclass
+class MelnorSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    state_fn: Callable[[Device], Any]
+
+
+@dataclass
+class MelnorSensorEntityDescription(
+    SensorEntityDescription, MelnorSensorEntityDescriptionMixin
+):
+    """Describes Melnor sensor entity."""
+
+    icon_fn: Callable[[Device], str] | None = None
+
+
+def _rssi_icon(device: Device) -> str:
+    """Return a representative icon based on signal strength."""
+
+    if device.rssi >= -20:
+        return "mdi:wifi-strength-4"
+    if device.rssi >= -40:
+        return "mdi:wifi-strength-3"
+    if device.rssi >= -60:
+        return "mdi:wifi-strength-2"
+    if device.rssi >= -80:
+        return "mdi:wifi-strength-1"
+    return "mdi:wifi-strength-outline"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
+
+    coordinator: MelnorDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors: list[MelnorSensorEntityDescription] = [
+        MelnorSensorEntityDescription(
+            device_class=SensorDeviceClass.BATTERY,
+            entity_category=EntityCategory.DIAGNOSTIC,
+            has_entity_name=True,
+            key="battery",
+            name="Battery",
+            native_unit_of_measurement=PERCENTAGE,
+            state_class=SensorStateClass.MEASUREMENT,
+            state_fn=lambda device: device.battery_level,
+        ),
+        MelnorSensorEntityDescription(
+            device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+            entity_category=EntityCategory.DIAGNOSTIC,
+            entity_registry_enabled_default=False,
+            has_entity_name=True,
+            icon_fn=_rssi_icon,
+            key="rssi",
+            name="RSSI",
+            native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+            state_class=SensorStateClass.MEASUREMENT,
+            state_fn=lambda device: device.rssi,
+        ),
+    ]
+
+    async_add_devices(
+        [
+            MelnorSensorEntity(
+                coordinator,
+                description,
+            )
+            for description in sensors
+        ],
+        True,
+    )
+
+
+class MelnorSensorEntity(MelnorBluetoothBaseEntity, SensorEntity):
+    """Representation of a Melnor sensor."""
+
+    entity_description: MelnorSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: MelnorDataUpdateCoordinator,
+        entity_description: MelnorSensorEntityDescription,
+    ) -> None:
+        """Initialize a sensor for a Melnor device."""
+        super().__init__(coordinator)
+
+        self.entity_description = entity_description
+
+        self._attr_unique_id = (
+            f"sensor-{self._attr_unique_id}-{self.entity_description.key}"
```

Comment:
I am not sure if adding the "sensor" prefix is needed here, but anyway you should avoid accessing `self._attr_unique_id` for reads.
```suggestion
            f"sensor-{super().unique_id}-{self.entity_description.key}"
```

Reply:
I don't think either of these are good practices, if we don't like how the super class sets the unique id, either:
1. rewrite it: here:
```py
 self._attr_unique_id = f"sensor-{self._device.mac}-{self.entity_description.key}"
 ```
2. 
Don't set self._attr_unique_id, implement a `unique_id` property method instead.

Reply:
I think what I'll do is set `self._attr_unique_id ` on the base class, and then on each entity I'll create a `unique_id` property to massage the unique value into the fully namespaced unique key for that entity. How does that sound?